### PR TITLE
prometheus-openldap-exporter: init at 2.1

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

See https://github.com/tomcz/openldap_exporter

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/monitoring/prometheus/openldap-exporter.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    -ldflags=
+      -X github.com/tomcz/openldap_exporter.tag=v${version}
+      -X github.com/tomcz/openldap_exporter.commit=unknown
+  '';
+
+  goPackagePath = "github.com/tomcz/openldap_exporter";
+
+  meta = with lib; {
+    homepage = "https://github.com/tomcz/openldap_exporter";
+    description = " This is a simple service that scrapes metrics from OpenLDAP and exports them via HTTP for Prometheus consumption";
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/openldap-exporter.nix:24]:
```suggestion
    description = "Simple service that scrapes metrics from OpenLDAP and exports them via HTTP for Prometheus consumption";
```

File: pkgs/servers/monitoring/prometheus/openldap-exporter.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ buildGoPackage, lib, fetchFromGitHub }:
+
+buildGoPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/openldap-exporter.nix:3]: We could use buildGoModule instead.

File: pkgs/servers/monitoring/prometheus/openldap-exporter.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "tomcz";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Di1GiyVp/hGCFhqxhlqJSucGZK7f/FDDUFtJRaiAZu4=";
+  };
+
+  buildFlagsArray = ''
+    -ldflags=
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/openldap-exporter.nix:15]:
```suggestion
    -ldflags=
      -s -w
```

File: pkgs/servers/monitoring/prometheus/openldap-exporter.nix (hunk 2 above)
Comment (Ma27) [reply pkgs/servers/monitoring/prometheus/openldap-exporter.nix:3]: I don't see a reason to use this fixed-output drv approach unless we have to though.

**Review by risicle — APPROVED** [review]
WFM, nixos test passes, x86_64

File: pkgs/servers/monitoring/prometheus/openldap-exporter.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/servers/monitoring/prometheus/openldap-exporter.nix:3]: As long as we don't generate a deps.nix it is whatever tbh.

**Review by SuperSandro2000 — APPROVED** [review]
package LGTM but dont know about the test or module

**Pull request merged**